modules/core/src/shaderlib/shadow/shadow.js

```diff
@@ -81,7 +88,7 @@ vec4 shadow_filterShadowColor(vec4 color) {
     float shadowAlpha = 0.0;
     for (int i = 0; i < max_lights; i++) {
       if(i < int(shadow_uLightCount)) {
-        shadowAlpha += shadow_getShadowWeight(shadow_vPosition[i], shadow_uShadowMap[i]) * shadow_uColor.a / shadow_uLightCount;
+        shadowAlpha += shadow_getShadowWeight(shadow_vPosition[i], i) * shadow_uColor.a / shadow_uLightCount;
```

Comment:
if statements are generally considered bad for shader performance. Since you only have two lights, might as well do

```glsl
shadowAlpha += shadow_getShadowWeight(shadow_vPosition[0], shadow_uShadowMap0);
shadowAlpha += shadow_getShadowWeight(shadow_vPosition[1], shadow_uShadowMap1);
shadowAlpha *= a;
```